test/origin_simulator/single_route_origin_simulator_test.exs

```diff
@@ -0,0 +1,235 @@
+    setup do
+      origin_payload = %{
+        "origin" => "https://www.bbc.co.uk/news",
+        "stages" => [%{"at" => 0, "status" => 200, "latency" => 0}],
+        "route" => "/news"
+      }
+
+      origin_payload_default_route = %{
+        "origin" => "https://www.bbc.co.uk/news",
+        "stages" => [%{"at" => 0, "status" => 200, "latency" => 0}]
+      }
+
+      {:ok,
+       default_route: OriginSimulator.Recipe.default_route(),
+       payload: origin_payload,
+       payload_default_route: origin_payload_default_route}
+    end
+
+    test "for a route", %{payload: payload} do
+      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
+      Process.sleep(20)
+
+      req_path = payload["route"]
+      conn = conn(:get, req_path)
+      conn = OriginSimulator.call(conn, [])
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert conn.resp_body == body_mock()
+    end
+
+    test "error message due to non-matching route", %{payload: payload} do
+      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
+      Process.sleep(20)
+
+      path = "/random_path"
+      conn = conn(:get, path)
+      conn = OriginSimulator.call(conn, [])
+
+      assert conn.state == :sent
+      assert conn.status == 406
+      assert conn.resp_body == recipe_not_set_message(path)
+    end
+
+    test "for default route", %{payload_default_route: payload, default_route: route} do
+      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
+      Process.sleep(20)
+
+      conn = conn(:get, route)
+      conn = OriginSimulator.call(conn, [])
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert conn.resp_body == body_mock()
+    end
+
+    test "for '/*' route", %{payload: payload} do
+      payload = Map.put(payload, "route", "/*")
+      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
+      Process.sleep(20)
+
+      conn = conn(:get, "/news/sport/id")
+      conn = OriginSimulator.call(conn, [])
+
+      assert conn.state == :sent
+      assert conn.status == 200
+      assert conn.resp_body == body_mock()
+
+      new_status = 500
+      new_stages = [%{"at" => 0, "status" => new_status, "latency" => 0}]
+      payload = Map.put(payload, "stages", new_stages)
+
+      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
+      Process.sleep(20)
+
+      conn = conn(:get, "/news/politcs/id")
+      conn = OriginSimulator.call(conn, [])
+
+      assert conn.state == :sent
+      assert conn.status == new_status
+      assert conn.resp_body == http_error_message(new_status)
+    end
+
+    test "for arbitrary wildcard route", %{payload: payload} do
+      payload = Map.put(payload, "route", "/news*")
+      conn(:post, "/add_recipe", Poison.encode!(payload)) |> OriginSimulator.call([])
+      Process.sleep(20)
+
+      conn = conn(:get, "/news/sport/id")
```

Comment:
We're unlikely to have two products combined, but I see it's an example of an arbitrary route.

Reply:
Using path of real news articles now.